### PR TITLE
Create a analytics for departments of store maps.

### PR DESCRIPTION
**Pull request by rodrigoulisses**
Increments department key for each get request to department in the api

### CODE REVIEW

File: spec/controllers/api/v2/maps_controller_spec.rb [hunk 1]
```diff
@@ -2,6 +2,22 @@
...
   describe 'GET #department' do
+    let!(:redis) { Redis.new(url: ENV['REDIS_URL']) }
+
+    context 'before get' do
+      it 'count departments is nil' do
+        expect(redis.get('100/sec12')).to eql nil
+      end
+    end
+
+    context 'after get' do
```
Comment (marclerodrigues) [inline spec/controllers/api/v2/maps_controller_spec.rb]: You can join these "before get" and "after get" in one test.
```ruby
it 'changes departments count' do
 expect do
  get :department, params: { store_id: 100, department_id: "12" }, format: :json
 end.to change(redis.get('100/sec12'))
end
```

File: app/business/map_analytics.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+class MapAnalytics
+  def initialize(key: nil, store: nil)
+    @key = key
+    @store = store
+    @redis = Redis.new(url: ENV['REDIS_URL'])
```
Comment (marclerodrigues) [inline app/business/map_analytics.rb]: You can move this `ENV['REDIS_URL']` to a constant.

File: spec/business/map_analytics_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.describe MapAnalytics do
+  let!(:key) { "8/sec4" }
+  let!(:store) { create(:store, leroy_id: 8) }
```
Comment (marclerodrigues) [inline spec/business/map_analytics_spec.rb]: I believe you don't need the bang in `key` and `store`.

File: spec/business/map_analytics_spec.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  describe '#list' do
+    context 'before call set!' do
+      it 'Ferramentas there is count 0' do
+        expect(subject.list).to include { name:"Ferramentas", count:0 }
+      end
+    end
+
+    context 'after call set!' do
```
Comment (marclerodrigues) [inline spec/business/map_analytics_spec.rb]: The same comment about trying to expect the change.

File: spec/business/map_analytics_spec.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+      described_class.set!(key: key)
+    end
+  end
+
+  describe '#set!' do
+    it 'increment count for index on redis' do
+      subject.set!
+
+      expect(redis.get(key)).to eql "1"
```
Comment (marclerodrigues) [inline spec/business/map_analytics_spec.rb]: The same comment about trying to expect the change.

**Comment by rodrigoulisses** [comment]
@marclerodrigues

**Pull request merged**